src/repository/posts.repository.js

```diff
@@ -0,0 +1,78 @@
+const { createPoolCluster } = require('mysql');
+const { connection } = require('../config/mysql');
+const { auth,db } = require('../config/firebase');
+const Post = require('../entities/post.entity');
+const { InvalidEntity } = require('../exceptions/InvalidEntity.exception');
+const { createPlaceholderString } = require('../utils/mysql.utils');
+
+function createPost(post){
+    return new Promise((resolve, reject)=>{
```

Comment:
please use `hasMissingKey` from 808067b62139b237aee91523800674bcd05bcaaf
to do validation on post object.

you can cherry-pick 808067b62139b237aee91523800674bcd05bcaaf . Then see the example in 2b4fb8a1f234db7ed9f638d190f5da15b92e7ec7 to know how to use it